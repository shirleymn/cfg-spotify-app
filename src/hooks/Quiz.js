import React, { useState } from 'react';

const QuizLogic = () => {
	const questions = [
		{
			questionText: 'This is Question 1?',
			answerOptions: [
				{ answerText: 'Answer 1', isCorrect: false },
				{ answerText: 'Answer 2', isCorrect: false },
				{ answerText: 'Answer 3', isCorrect: true },
				{ answerText: 'Answer 4', isCorrect: false },
			],
		},
		{
			questionText: 'This is Question 2?',
			answerOptions: [
				{ answerText: 'Answer_1', isCorrect: false },
				{ answerText: 'Answer_2', isCorrect: true },
				{ answerText: 'Answer_3', isCorrect: false },
				{ answerText: 'Answer_4', isCorrect: false },
			],
		},
		
	];

    const [currentQuestion, setCurrentQuestion] = useState(0);

    const [showScore, setShowScore] = useState(false);

    const [score, setScore] = useState(0)
    // Anytime the user clicks a button (selects an answer) the question number will be incremented and the user will go onto the next question
    const handleAnswerButtonClick = (isCorrect) => {
        if(isCorrect===true){
            setScore(score + 1);
        }

        const nextQuestion = currentQuestion + 1;
        if(nextQuestion < questions.length){
            setCurrentQuestion(nextQuestion);
        } else{
            setShowScore(true) // once the user has reached the end of the quiz the score panel will be displayed to the user by changing the set variable setShowScore from false to true causing react to re-render
        }
    }


	return (
		
		<div className='app'>
			{showScore ? (
				<div className='score-section'>You scored {score} out of {questions.length}</div>
			) : (
				<>
					<div className='question-section'>
						<div className='question-count'>
							<span>Question {currentQuestion+1} </span>/{questions.length}
						</div>
						<div className='question-text'>{questions[currentQuestion].questionText}</div>
					</div>
					<div className='answer-section'>
						{questions[currentQuestion].answerOptions.map((answerOptions) => 
                            <button onClick = {() => handleAnswerButtonClick(answerOptions.isCorrect)}>{answerOptions.answerText}</button>)}
					</div>
				</>
			)}
		</div>
	);
}

export default QuizLogic;