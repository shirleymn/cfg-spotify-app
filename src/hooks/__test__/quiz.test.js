import Quiz from "../../pages/quiz"

import React from 'react';

import renderer from 'react-test-renderer';
import { render, screen} from '@testing-library/react';
it('renders correctly', () => {
 const tree = renderer
   .create(<Quiz page="http://localhost:3000/quiz" />)
   .toJSON();
   console.log(tree)
   expect(tree).toMatchSnapshot();
});

test('renders each react link', () => {
  render(<Quiz/>);
  const linkElement = screen.getByText(/quiz/i)
  expect(linkElement).toBeInTheDocument();
});

// test('shows proper heading when rendered', () => {
//   render(<Quiz/>);
//   const {getByText} = render {name:"quiz-heading"})

//   expect(getByText('Discover how in tune you are with your music taste')).toBeInTheDocument()
// }

