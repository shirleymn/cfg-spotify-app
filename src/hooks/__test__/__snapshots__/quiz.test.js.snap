// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="Content"
  >
    <h1
      className="quiz-title"
    >
      Quiz
    </h1>
    <p
      className="quiz-subheading"
    >
      Discover how in tune you are with your music taste
    </p>
    <div
      className="question-text"
    >
      Who was your top artist in the last 4 weeks?
    </div>
    <ul
      className="Answers"
      disabled={false}
    >
      <li
        className=""
        onClick={[Function]}
      >
        Kanye West
      </li>
      <li
        className=""
        onClick={[Function]}
      >
        Ella Mai
      </li>
      <li
        className=""
        onClick={[Function]}
      >
        Kelhani
      </li>
      <li
        className=""
        onClick={[Function]}
      >
        Beyonce
      </li>
    </ul>
    <div
      className="answer-output"
    />
    <svg
      className="next-question"
      disabled={true}
      fill="currentColor"
      height="1em"
      onClick={[Function]}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 14a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V2a2 2 0 0 0-2-2H2a2 2 0 0 0-2 2v12zm4.5-6.5h5.793L8.146 5.354a.5.5 0 1 1 .708-.708l3 3a.5.5 0 0 1 0 .708l-3 3a.5.5 0 0 1-.708-.708L10.293 8.5H4.5a.5.5 0 0 1 0-1z"
      />
    </svg>
  </div>
</div>
`;
